### PR TITLE
App services upgrade with logs output to the console

### PR DESCRIPTION
**Pull request by jonalmeida**
For testing only.

### CODE REVIEW

**Comment by jonalmeida** [comment]
Weird. It doesn't seem to be failing now with the gradle issue. The current failure is from a network issue in the linter.

Link to TC logs: https://firefox-ci-tc.services.mozilla.com/tasks/groups/TUWETGKHQ7ulGWiQgkr4lw

**Comment by csadilek** [comment]
Just adding labels so this doesn't land by mistake :), see https://github.com/mozilla-mobile/android-components/pull/9177#issuecomment-742126983 for a new startup crash in 68.0.0.

**Comment by jonalmeida** [comment]
Oh, ha! The gradle build is failing, but the result from the last function returns 0, so TC thinks the tasks passes:

```
[task 2020-12-10T03:28:04.432Z] warn: removing resource mozilla.components.browser.tabstray.test:string/mozac_browser_errorpages_page_title without required default value.
[task 2020-12-10T03:28:04.432Z] 
[task 2020-12-10T03:28:04.432Z] ----- End of the daemon log -----
[task 2020-12-10T03:28:04.432Z] 
[task 2020-12-10T03:28:04.432Z] JVM crash log found: file:///builds/worker/checkouts/src/hs_err_pid1671.log
[task 2020-12-10T03:28:04.437Z] 
[task 2020-12-10T03:28:04.437Z] FAILURE: Build failed with an exception.
[task 2020-12-10T03:28:04.437Z] 
[task 2020-12-10T03:28:04.437Z] * What went wrong:
[task 2020-12-10T03:28:04.437Z] Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)
[task 2020-12-10T03:28:04.437Z] 
[task 2020-12-10T03:28:04.437Z] * Try:
[task 2020-12-10T03:28:04.437Z] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[task 2020-12-10T03:28:04.437Z] 
[task 2020-12-10T03:28:04.437Z] * Get more help at https://help.gradle.org
[task 2020-12-10T03:28:04.437Z] > Task :browser-engine-gecko:packageDebugAndroidTest
[task 2020-12-10T03:28:04.437Z] > Task :browser-engine-gecko-beta:packageDebugAndroidTest
[task 2020-12-10T03:28:04.437Z] > Task :browser-storage-sync:mergeDebugAndroidTestJavaResource
[task 2020-12-10T03:28:04.437Z] > Task :browser-engine-gecko-nightly:packageDebugAndroidTest
[task 2020-12-10T03:28:04.437Z] > Task :browser-state:packageDebugAndroidTest
[task 2020-12-10T03:28:04.437Z] > Task :browser-storage-memory:packageDebugAndroidTest
[task 2020-12-10T03:28:04.438Z] > Task :browser-tabstray:checkDebugAndroidTestDuplicateClasses
[task 2020-12-10T03:28:04.438Z] > Task :browser-storage-sync:mergeExtDexDebugAndroidTest
[task 2020-12-10T03:28:04.758Z] + copy_gradle_crash_log_and_exit
[task 2020-12-10T03:28:04.758Z] + mkdir -p /builds/worker/artifacts
[task 2020-12-10T03:28:04.766Z] ++ find /builds/worker/checkouts/src -name '*.log'
[task 2020-12-10T03:28:54.489Z] + logs=/builds/worker/checkouts/src/hs_err_pid1671.log
[task 2020-12-10T03:28:54.490Z] + '[' /builds/worker/checkouts/src/hs_err_pid1671.log. '!=' . ']'
[task 2020-12-10T03:28:54.490Z] + cp /builds/worker/checkouts/src/hs_err_pid1671.log /builds/worker/artifacts
[task 2020-12-10T03:28:54.492Z] # Some tests may be flaky, although they still download dependencies. So we let the following
[task 2020-12-10T03:28:54.492Z] # command fail, if needed.
[task 2020-12-10T03:28:54.492Z] set +e; ./gradlew $GRADLE_ARGS -Pcoverage $TEST_COMMANDS; set -e
[task 2020-12-10T03:28:54.492Z] + set +e
```

From the [`hs_err_pid1671.log`](https://firefoxci.taskcluster-artifacts.net/dHqLKz2GQliGXAXIn51ekQ/0/public/build/hs_err_pid1671.log) file, it does seem like we're running out of memory with 194MB left:
```
Memory: 4k page, physical 15842708k(198832k free), swap 0k(0k free)
```


Link to TC logs: https://firefox-ci-tc.services.mozilla.com/tasks/YRLQ9bZST2moVA_7o4eB7A

**Comment by pocmo** [comment]
> ```
> Memory: 4k page, physical 15842708k(198832k free), swap 0k(0k free)
> ```
> 
> Link to TC logs: https://firefox-ci-tc.services.mozilla.com/tasks/YRLQ9bZST2moVA_7o4eB7A

I recently updated `gradle.properties` to be closer to the androidx configuration. This also included a change about how much memory we assign to gradle:
https://github.com/mozilla-mobile/android-components/blob/master/gradle.properties#L12

If you dial this back to around `4096m` as before. Does that help?

**Comment by jonalmeida** [comment]
> If you dial this back to around `4096m` as before. Does that help?

@pocmo Yeah, that seems to have worked! I'm going make this change on the https://github.com/mozilla-mobile/android-components/pull/9179 PR which we're going to land.

Link to TC logs: https://firefox-ci-tc.services.mozilla.com/tasks/YW4h7xbqRG2rCq6BxLB8ug

**Comment by pocmo** [comment]
> @pocmo Yeah, that seems to have worked! I'm going make this change on the #9179 PR which we're going to land.

r+ :)

**Comment by jonalmeida** [comment]
Closing this test PR now.

File: taskcluster/scripts/toolchain/android-gradle-dependencies.sh [hunk 1]
```diff
@@ -11,6 +11,15 @@ function get_abs_path {
...
 }
 
+function copy_gradle_crash_log_and_exit {
+    # XXX unhardcode me
+    mkdir -p /builds/worker/artifacts
+    logs=`find /builds/worker/checkouts/src -name \*.log`
+    if [ "${logs}." != "." ] ; then
+        cp $logs /builds/worker/artifacts
+    fi
+}
```
Comment (escapewindow) [inline taskcluster/scripts/toolchain/android-gradle-dependencies.sh:21]: Oops, I had an `exit 1` in my test script but may have dropped it here.

**Pull request closed without merging**